hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/utils/KeyActivationUtils.java

```diff
@@ -58,25 +61,31 @@ private KeyActivationUtils() {
 	 * (as needed for e.g. the {@link TransferLogic} implementation), we can assume the target address
 	 * is a mirror address. All other addresses we resolve to their mirror form before proceeding.
 	 *
-	 * @param frame
-	 * 		current frame
-	 * @param target
-	 * 		the element to test for key activation, in standard form
-	 * @param activationTest
-	 * 		the function which should be invoked for key validation
-	 * @param ledgers the current Hedera world state
-	 * @param aliases the current Hedera contract aliases
+	 * @param provider       provides the current frame or precompile message
+	 * @param target         the element to test for key activation, in standard form
+	 * @param activationTest the function which should be invoked for key validation
+	 * @param ledgers        the current Hedera world state
+	 * @param aliases        the current Hedera contract aliases
 	 * @return whether the implied key is active
 	 */
 	public static boolean validateKey(
-			final MessageFrame frame,
+			final PrecompileInfoProvider provider,
 			final Address target,
 			final KeyActivationTest activationTest,
 			final WorldLedgers ledgers,
-			final ContractAliases aliases
+			final Optional<ContractAliases> aliases
```

Comment:
This seems workable, but I would greatly prefer to revert the changes to this method, and create a new method especially for direct calls,

```
public static boolean validateKeyForDirectCall(
    final DirectCallsPrecompileInfoProvider directProvider,
    final Address target.
    final KeyActivationTest activationTest,
    final WorldLedgers ledger
) {
```

Reply:
Thanks for the detailed review and comments. However, I'm not sure that I understood this one correctly. Does the last commit simplify the logic for validateKey enough, or the aim here is to explicitly check every precompile impl. for directToken logic and call either validateKeyForDirectCall or validateKey(MessageFrame...)?

Reply:
Np! I like the step you took...could we go further by adding a method to `PrecompileInfoProvider`,
```
   ...
   @Nullable
   ContractAliases aliases();
```
...and then have,
```
	public static boolean validateKey(
			final PrecompileInfoProvider precompileInfoProvider,
			final Address target,
			final KeyActivationTest activationTest,
			final WorldLedgers ledgers,
			@Nullable final ContractAliases contractAliases
	) {
		if (precompileInfoProvider.isDirectTokenCall()) {
			return validateKeyForDirectCall((DirectCallsPrecompileInfoProvider) precompileInfoProvider,
					target, activationTest, ledgers);
		}
                validateTrue(contractAliases != null, FAIL_INVALID);
                return validateKeyForCallWithMessageFrame(frame, target, activationTest, ledgers, aliases);
	}
```
Imo this would make both this method _and_ the client code that calls it cleaner.